lib/scanner/index.js

```diff
@@ -43,9 +57,44 @@ class Scanner extends Emitter {
   stop() {
     debug('stop');
     MagnetScanner.stop();
+    this.stopExpireCheck();
     this.listener.remove();
   }
 
+  startExpireCheck() {
+    this.nextExpireCheck = setTimeout(() => {
```

Comment:
Should we do the `if (!this.expiryEnabled()) return;` here so we avoid adding the setTimeout and the execution of code that later on is going to return?